packages/kit/src/DropdownButton.tsx

```diff
@@ -1,70 +1,56 @@
-import React from 'react'
-import { Button, Sheet, YStack } from 'tamagui'
-import { z } from 'zod'
+/** @jsxImportSource @final-ui/react */
 
-const DropdownButtonProps = z.object({
-  id: z.string().optional(),
-  onSelect: z.function().args(z.string()).optional(),
-  buttonProps: z.any().optional(),
-  options: z
-    .array(
-      z.object({
-        key: z.string(),
-        label: z.string(),
-        icon: z.any().optional(),
-      })
-    )
-    .optional(),
-})
+import { Only, setStateAction, state, StateDataState, WithServerProps } from '@final-ui/react'
+import {
+  Button,
+  Sheet,
+  SheetFrame,
+  SheetHandle,
+  SheetOverlay,
+  YStack,
+} from '@final-ui/tamagui/server'
 
-export function DropdownButton(props: z.infer<typeof DropdownButtonProps>) {
-  const { options } = props
-  const [open, setOpen] = React.useState(false)
+type Props<T = string> = {
+  value: StateDataState<T>
+  onSelect?: (item: string) => void
+  button?: JSX.Element
+  options?: {
+    key: Only<T>
+    label: string
+  }[]
+}
+
+export function DropdownButton(props: WithServerProps<Props>) {
+  const isOpen = state(false)
   return (
-    <>
-      <Button
-        {...props.buttonProps}
-        onPress={() => {
-          setOpen(true)
-        }}
-      />
+    <YStack>
+      <Button onPress={setStateAction(isOpen, true)}>{props.button}</Button>
       <Sheet
-        forceRemoveScrollEnabled={open}
+        forceRemoveScrollEnabled={isOpen}
         modal={true}
-        open={open}
-        onOpenChange={setOpen}
-        // snapPoints={[50]}
-        // snapPointsMode={snapPointsMode}
+        open={isOpen}
+        onOpenChange={setStateAction(isOpen)}
         dismissOnSnapToBottom
-        // position={position}
-        // onPositionChange={setPosition}
         zIndex={100_000}
         animation="medium"
       >
-        <Sheet.Overlay animation="quick" enterStyle={{ opacity: 0 }} exitStyle={{ opacity: 0 }} />
-        <Sheet.Handle />
-        <Sheet.Frame padding="$4" justifyContent="center" space="$5">
+        <SheetOverlay animation="quick" enterStyle={{ opacity: 0 }} exitStyle={{ opacity: 0 }} />
+        <SheetHandle />
+        <SheetFrame padding="$4" justifyContent="center" space="$5">
           <YStack gap="$3">
-            {options?.map((item) => {
+            {props.options?.map((item) => {
               return (
                 <Button
                   key={item.key}
-                  onPress={() => {
-                    setOpen(false)
-                    props.onSelect?.(item.key)
-                  }}
+                  onPress={[setStateAction(props.value, item.key), setStateAction(isOpen, false)]}
```

Comment:
This is not ideal (we should accept prop instead), but currently we do not have a way to change `event.payload`. `onPress` does not have any arguments, so `props.onSelect` will not work. It works for `SelectField`, because `onValueChange` already calls with the current value.

We may need to create at some point later a way to write the following:
```
onPress{bind(props.onSelect, item.key)}
```
or something.